src/sql/workbench/parts/dataExplorer/electron-browser/nodeActions.contribution.ts

```diff
@@ -15,27 +15,83 @@ MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
 		id: DISCONNECT_COMMAND_ID,
 		title: localize('disconnect', 'Disconnect')
 	},
-	when: NodeContextKey.IsConnected
+	when: ContextKeyExpr.and(
+		new ContextKeyDefinedExpr('isConnected'),
+		new ContextKeyEqualsExpr('isConnected', true))
 });
 
+// For Database nodes under a Server in the SQL Servers folder
 MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
 	group: 'connection',
 	order: 2,
 	command: {
 		id: NEW_QUERY_COMMAND_ID,
 		title: localize('newQuery', 'New Query')
 	},
-	when: NodeContextKey.IsConnectable
+	when: ContextKeyExpr.and(
```

Comment:
Thoughts on adding an Or expression? Then I wouldn't need to define a new entry for each viewItem type which would be awesome...

Reply:
We do have an or expr that kevin added

Reply:
Where? All I'm seeing is >= and <= @kevcunnane ?
```
export const enum ContextKeyExprType {
	Defined = 1,
	Not = 2,
	Equals = 3,
	NotEquals = 4,
	And = 5,
	Regex = 6,
	// {{SQL CARBON EDIT}}
	GreaterThanEquals = 7,
	LessThanEquals = 8
	//
}
```

Reply:
I must have thought he did. nvm

Reply:
VSCode folks pushed back against adding "or" to their support. Apparently it's harder to get the logic right here. Just FYI - not saying we should't, but frankly we should consider pushing this back upstream to them whatever we do.

Reply:
I'm less concerned about it now since we can just use regexes, if we feel this would provide other use though I'm willing to investigate doing that later on.